spec/index.bs

```diff
@@ -5131,70 +5144,114 @@ enum GPUStoreOp {
             <pre class=argumentdef for="GPURenderEncoderBase/setPipeline(pipeline)">
-                pipeline:
+                |pipeline|: The render pipeline to use for subsequent drawing commands.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setPipeline()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |pipeline| is [=valid=].
+                        - |pipeline|.{{GPUObjectBase/[[device]]}} is |this|.
+                    </div>
+                1. Set |this|.{{GPURenderEncoderBase/[[pipeline]]}} to be |pipeline|.
+            </div>
         </div>
 
     : <dfn>setIndexBuffer(buffer, indexFormat, offset, size)</dfn>
     ::
+        Sets the current index buffer.
 
         <div algorithm="GPURenderEncoderBase.setIndexBuffer">
             **Called on:** {{GPURenderEncoderBase}} this.
 
             **Arguments:**
             <pre class=argumentdef for="GPURenderEncoderBase/setIndexBuffer(buffer, indexFormat, offset, size)">
-                buffer:
-                indexFormat:
-                offset:
-                size:
+                |buffer|: Buffer containing index data to use for subsequent drawing commands.
+                indexFormat: Format of the index data contained in |buffer|.
+                |offset|: Offset in bytes into |buffer| where the index data begins.
+                |size|: Size in bytes of the index data in |buffer|.
+                    If `0`, |buffer|.{{GPUBuffer/[[size]]}} - |offset| is used.
             </pre>
 
             **Returns:** void
 
-            Issue: Describe {{GPURenderEncoderBase/setIndexBuffer()}} algorithm steps.
+            Issue the following steps on the [=Device timeline=] of |this|:
+            <div class=device-timeline>
+                1. If any of the following conditions are unsatisfied, generate a
+                    {{GPUValidationError}} error and stop.
+                    <div class=validusage>
+                        - |buffer| is [=valid=].
+                        - |buffer|.{{GPUObjectBase/[[device]]}} is |this|.
+                        - |offset| &le; |buffer|.{{GPUBuffer/[[size]]}}.
+                        - |size| &le; |buffer|.{{GPUBuffer/[[size]]}} - |offset|.
+                    </div>
+                1. Add |buffer| to the [=usage scope=] as
+                    {{GPUBufferUsage/INDEX|GPUBufferUsage.INDEX}}).
+                1. Set |this|.{{GPURenderEncoderBase/[[index_buffer]]}} to be |buffer|.
+            </div>
         </div>
 
     : <dfn>setVertexBuffer(slot, buffer, offset, size)</dfn>
     ::
+        Sets the current vertex buffer for a given bind group slot.
```

Comment:
Vertex buffers have a separate/unrelated binding table from bind groups. (Note we currently try to use the word "slot" for vertex buffers and "index" for bind groups, but I don't think we're consistent yet. We could possibly come up with better words too.) [🦈 ]

Reply:
Leaving this as "slot" for now, but fixed the erroneous "bind group" mentions. Don't have any suggestions for better verbiage.